server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/BasicAuthenticationTokenManager.scala

```diff
@@ -0,0 +1,40 @@
+package org.apache.james.jmap.http
+
+import java.util.Base64
+
+import javax.inject.Inject
+import org.apache.james.core.Username
+import org.apache.james.jmap.http.BasicAuthenticationTokenManager.PREFIX
+import org.apache.james.user.api.UsersRepository
+
+object BasicAuthenticationTokenManager {
+  val PREFIX: String = "Basic "
+}
+
+class BasicAuthenticationTokenManager @Inject() (val usersRepository: UsersRepository) {
+  class UserRequest(val username: String, val password: String)
+
+  private def userRequestExtraction(token: String): UserRequest = {
+    val arr: Array[String] = token.split(":")
+    if (arr.length != 2) {
+      throw new IllegalArgumentException("BasicAuthentication invalid")
+    }
+
+    new UserRequest(arr(0), arr(1))
+  }
+
+  private def validUsername(username: String): Boolean = usersRepository.contains(Username.of(username))
+
+  private def validPassword(password: String, username: String): Boolean = usersRepository.test(Username.of(username), password)
+
+  private def tokenDecoder(token: String): UserRequest = userRequestExtraction(new String(Base64.getDecoder.decode(token)))
+
+  def isValid(token: String): Boolean = {
```

Comment:
We should IMO have a case class to strong type this `token: String`, then once we have a strong type we could add some behaviours to it, like base64 decoding and UserRequest parsing.